### ISSUE TITLE
Partial failure during object overwrite causes information corruption in meta2 database

### ISSUE THREAD

**Issue by sebastienlapierre** [open-io/oio-sds#949]
__Configuration__
The namespace is configured to put
* 2 meta2 databases on Site1 and 1 database on Site2
* 2 chunks on Site1 and 1 chunk on Site2, or all chunks on the only responding site

__Issue__
Site2 is stopped.
If we try to overwrite an existing object, we get this error:
```openio object create --oio-ns NAMESPACE --oio-account TEST mytestcontainer /etc/magic
Failed to connect to {u'url': u'http://172.20.11.3:6043/FD047813ECA40DB1044740A196A5CD815E45864D512193F9A49EAFA213D80A4C', u'score': 96, u'hash': u'00000000000000000000000000000000', u'pos': '0', u'size': 536870912} ([Errno 111] ECONNREFUSED)
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/oio/api/replication.py", line 94, in _connect_put
    parsed.netloc, 'PUT', parsed.path, h)
  File "/usr/lib/python2.7/dist-packages/oio/common/http.py", line 127, in http_connect
    conn.endheaders()
  File "/usr/lib/python2.7/httplib.py", line 1035, in endheaders
    self._send_output(message_body)
  File "/usr/lib/python2.7/httplib.py", line 879, in _send_output
    self.send(msg)
  File "/usr/lib/python2.7/httplib.py", line 841, in send
    self.connect()
  File "/usr/lib/python2.7/dist-packages/oio/common/http.py", line 73, in connect
    r = HTTPConnection.connect(self)
  File "/usr/lib/python2.7/httplib.py", line 822, in connect
    self.timeout, self.source_address)
  File "/usr/lib/python2.7/dist-packages/eventlet/green/socket.py", line 63, in create_connection
    raise err
error: [Errno 111] ECONNREFUSED
+-------+------+----------------------------------+
| Name  | Size | Hash                             |
+-------+------+----------------------------------+
| magic |  111 | 272913026300E7AE9B5E2D51F138E674 |
+-------+------+----------------------------------+
``` 

Object information in meta2 database:
```
openio object locate --oio-ns NAMESPACE --oio-account TEST mytestcontainer magic
+-----+------------------------------------------------------------------------------------------+-----------+----------------------------------+
| Pos | Id                                                                                       |      Size | Hash                             |
+-----+------------------------------------------------------------------------------------------+-----------+----------------------------------+
| 0   | http://172.20.0.81:6026/C5DA5B1AAD6FEE18C2E0266D091553B296EE6161B433C1DCCDAF6824BAA770E2 |       111 | 272913026300E7AE9B5E2D51F138E674 |
| 0   | http://172.20.0.11:6027/070A154742BE4AE223FEBF2C7B515E8F52B99D15AD1904BCD0B5A3486F065E7B |       111 | 272913026300E7AE9B5E2D51F138E674 |
| 0   | http://172.20.11.3:6043/FD047813ECA40DB1044740A196A5CD815E45864D512193F9A49EAFA213D80A4C | 536870912 | 00000000000000000000000000000000 |
+-----+------------------------------------------------------------------------------------------+-----------+----------------------------------+
``` 

Information about chunk http://172.20.11.3:6043/FD047813ECA40DB1044740A196A5CD815E45864D512193F9A49EAFA213D80A4C is erroneous.

**Comment by jfsmig** [comment]
Fixed in #925 and #919

**Comment by fvennetier** [comment]
What happened:
* The failure of Site2 has probably taken some time to be reflected by service scores.
* Before the score has changed, an upload has been started, with one chunk on Site2.
* The chunk on Site2 could not be uploaded (see the stack trace).
* **The chunk on Site2 has still been referenced in meta2 database** (this should be fixed by #925).
